### PR TITLE
Add logic for downloading map data into database

### LINKED ISSUE #21: Add logic for downloading map data into database
# Overview
Using a Ktor client, implement a GET request that downloads some JSON data. 

# Technical Details

This function should be encapsulated in a class. The class should have two constructor properties: one for the `Database`, and one for an `HttpClient`. These properties will be supplied by other parts of the code later on.

The server endpoint that the request will be made to should be configured via the `local.properties` file similar to how our mapbox keys are currently set up. Add a string field to the `buildkonfig` setup to read in some url. For the scope of this task, it doesn't matter what this url is since we won't be testing with an actual server.

The returned JSON data will be a `FeatureCollection` following the GeoJson standard. You can use the [Spatial K library](https://dellisd.github.io/spatial-k/geojson/#from-json) to convert this JSON text into an object.

Using the queries and table created in #13, insert each `Feature` into the table by mapping each Spatial K feature to a database Feature. Here's how the features from the FeatureCollection map to the database for now:

| Name | Mapping |
|--------|--------|
|`fid`|Get from `Feature` properties|
|`common_name`|Use `name` property from props |
|`secondary_name`|Set to `null` for now|
|`type`|Use `type` property from props|
|`building`|Use `building` property from props, or `null` if not available|
|`floor`|Use `floor` property from props, or `null` if not available| 
|`search_tags`|Use the `name` property from props for now|
|`json`|Convert the `Feature` [back to a string](https://dellisd.github.io/spatial-k/geojson/#to-json). (This will be done by an adapter once #18 is complete.)|

I recommend creating a separate function for this. An extension function would be good for this purpose.

# Testing

Using [Ktor's mock client(s)](https://ktor.io/clients/http-client/testing.html), write tests to ensure that data is correctly downloaded and inserted into the database. I recommend waiting until #18 is finished so that you can reuse some of the database test code.

### PR DESCRIPTION
**Pull request by safa-howaid**
`server.url=<random url>` should be added in the `local.properties` file for the build to run successfully.'

Resolves #21

### CODE REVIEW

File: .github/workflows/gradle-check.yml [hunk 1]
```diff
@@ -27,6 +27,8 @@ jobs:
       env:
         MAPBOX_KEY: ${{ secrets.MAPBOX_KEY }}
       run: echo mapbox.key="$MAPBOX_KEY" > local.properties
+    - name: Configure server endpoint
+      run: echo server.url=https://example.com/ > local.properties
```
Comment (dellisd) [inline .github/workflows/gradle-check.yml:31]:
```suggestion
      env:
        MAPBOX_KEY: ${{ secrets.MAPBOX_KEY }}
      run: |
        echo mapbox.key="$MAPBOX_KEY" > local.properties
        echo server.url=https://example.com/ > local.properties
```
Fancy multi-line yml syntax

File: common/src/androidTest/kotlin/FeatureApiTest.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    @Test
+    fun `check that data is downloaded and inserted to database`() {
+        runBlocking {
+            FeatureApi(database, client).getAndStoreFeatures()
+        }
+        assertEquals(database.featureQueries.getAll().executeAsList(),featureList)
+    }
+}
```
Comment (dellisd) [inline common/src/androidTest/kotlin/FeatureApiTest.kt:93]:
```suggestion
    fun `check that data is downloaded and inserted to database`() = runBlocking {
        FeatureApi(database, client).getAndStoreFeatures()
       
        assertEquals(database.featureQueries.getAll().executeAsList(),featureList)
    }
}
```

Minor style change

File: common/src/commonMain/kotlin/com/cuhacking/atlas/common/FeatureApi.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.cuhacking.atlas.db.Database
+import io.github.dellisd.spatialk.geojson.Feature
+import io.github.dellisd.spatialk.geojson.FeatureCollection.Companion.toFeatureCollection
+import io.ktor.client.*
+import io.ktor.client.request.*
+
+class FeatureApi (private val database: Database, private val client: HttpClient) {
+
+    private fun Feature.toDataFeature() = com.cuhacking.atlas.db.Feature(
```
Comment (dellisd) [inline common/src/commonMain/kotlin/com/cuhacking/atlas/common/FeatureApi.kt:11]:
```suggestion
    private fun Feature.toDataFeature() = DbFeature(
```
You can import this as an alias.
```kotlin
import com.cuhacking.atlas.db.Feature as DbFeature
```

File: common/src/commonMain/kotlin/com/cuhacking/atlas/common/FeatureApi.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        properties["id"].toString().toLong(),
+        properties["name"].toString().replace("\"", ""),
+        null,
+        properties["type"].toString().replace("\"", ""),
+        properties["building"].toString().replace("\"", ""). takeIf { it != "null" },
+        properties["floor"].toString().replace("\"", ""). takeIf { it != "null" },
+        properties["name"].toString().replace("\"", ""),
+        this)
+
+    suspend fun getAndStoreFeatures() {
```
Comment (dellisd) [inline common/src/commonMain/kotlin/com/cuhacking/atlas/common/FeatureApi.kt:21]:
```suggestion
    suspend fun getAndStoreFeatures() = withContext(dispatchers.io) {
```
We want our network requests to run on the I/O thread

**Review by dellisd — CHANGES_REQUESTED** [review]

**Review by dellisd — APPROVED** [review]

**Review by ghaskins99 — APPROVED** [review]

**Pull request merged**